cli/src/cmd/deploy.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{
+    cmd::{
+        build,
+        Result,
+    }
+};
+
+use futures::future::Future;
+use runtime_primitives::{
+    generic::{Era, Header},
+    traits::{IdentityLookup, Verify, BlakeTwo256},
+    AnySignature,
+};
+use std::{
+    collections::HashMap,
+    fs,
+    io::Read,
+    path::{Path, PathBuf},
+};
+use substrate_primitives::{
+    crypto::Pair,
+    sr25519,
+    H256,
+};
+use subxt::{
+    balances::Balances,
+    contracts::{
+        Contracts,
+        ContractsXt,
+    },
+    system::System,
+};
+
+/// Load the wasm blob from the specified path, or defaults to the target contract wasm in the
+/// current project, inferred via the crate metadata.
+fn load_contract_code(path: Option<&PathBuf>) -> Result<Vec<u8>> {
+    let default_wasm_path = build::collect_crate_metadata()
+        .map(|metadata| metadata.dest_wasm())?;
+    let contract_wasm_path = path.unwrap_or(&default_wasm_path);
+
+    let mut data = Vec::new();
+    let mut file = fs::File::open(&contract_wasm_path)
+        .map_err(|e| format!("Failed to open {}: {}", contract_wasm_path.display(), e))?;
+    file.read_to_end(&mut data)?;
+
+    return Ok(data)
+}
+
+/// If the extrinsic was successfully included in a block, attempt to extract the code hash from the
+/// `Contracts::CodeStored` event
```

Comment:
Normally you would want to write doc-comments as

----
Short description of what it does on a high-level.

Detailed and longer more elaborate description of what it does with mentions of exceptions and potential failures.
----

Reply:
Yeah I've change this comment, not sure it is an improvement. Though that method is private and very succinct anyway